rslib/src/backend/mod.rs

```diff
@@ -137,10 +137,46 @@ impl Backend {
             .clone()
     }
 
-    fn web_client(&self) -> &Client {
+    #[cfg(feature = "rustls")]
+    fn set_custom_certificate_inner(&self, cert_str: String) -> Result<()> {
+        use std::io::Cursor;
+        use std::io::Read;
+
+        use reqwest::Certificate;
+
+        let mut web_client = self.web_client.lock().unwrap();
+
+        if cert_str.is_empty() {
+            let _ = web_client.insert(Client::builder().http1_only().build().unwrap());
+            return Ok(());
+        }
+
+        if rustls_pemfile::read_all(Cursor::new(cert_str.as_bytes()).by_ref()).count() != 1 {
+            return Err(AnkiError::InvalidCertificateFormat);
+        }
+
+        if let Ok(certificate) = Certificate::from_pem(cert_str.as_bytes()) {
+            if let Ok(new_client) = Client::builder()
+                .use_rustls_tls()
+                .add_root_certificate(certificate)
+                .http1_only()
+                .build()
+            {
+                let _ = web_client.insert(new_client);
+                return Ok(());
+            }
+        }
+
+        Err(AnkiError::InvalidCertificateFormat)
+    }
+
+    fn web_client(&self) -> Client {
         // currently limited to http1, as nginx doesn't support http2 proxies
-        self.web_client
-            .get_or_init(|| Client::builder().http1_only().build().unwrap())
+        let mut web_client = self.web_client.lock().unwrap();
+
+        return web_client
+            .get_or_insert(Client::builder().http1_only().build().unwrap())
```

Comment:
Please use get_or_insert_with() so that a new client is not created when one already exists.